src/main/java/pl/wojciechkostecki/revel/service/MenuService.java

```diff
@@ -0,0 +1,61 @@
+package pl.wojciechkostecki.revel.service;
+
+import org.springframework.stereotype.Service;
+import pl.wojciechkostecki.revel.exception.MenuIsAlreadyAssignedException;
+import pl.wojciechkostecki.revel.mapper.MenuMapper;
+import pl.wojciechkostecki.revel.model.Local;
+import pl.wojciechkostecki.revel.model.Menu;
+import pl.wojciechkostecki.revel.model.MenuItem;
+import pl.wojciechkostecki.revel.model.dto.MenuDTO;
+import pl.wojciechkostecki.revel.repository.MenuRepository;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@Transactional
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final LocalService localService;
+    private final MenuMapper menuMapper;
+
+    public MenuService(MenuRepository menuRepository, LocalService localService, MenuMapper menuMapper) {
+        this.menuRepository = menuRepository;
+        this.localService = localService;
+        this.menuMapper = menuMapper;
+    }
+
+    public Menu save(MenuDTO menuDTO) {
+        Menu menu = menuMapper.toEntity(menuDTO);
+        Local local = localService.findById(menuDTO.getLocalId());
```

Comment:
You could do the same thing that Ive said above. Use the mapper and you wont have to retrieve the local here.